### DISCUSSION TITLE
Client QA Summary

### DISCUSSION THREAD

**Discussion by KingSkyLi** [TuGraph-family/tugraph-db#109] [Q&A]
本贴是客户端的QA汇总贴，详见下方回答。
This post is the QA summary post of the client, see the answer below for details.

**Comment by KingSkyLi** [comment]
中文：
Q：client 目前有哪些编程语言，是否支持 node js？
A：目前主要支持的编程语言有 c++,python,java；目前不支持 node js。使用 node 作为主要开发语言的用户，可以使用 tugraph 提供的 restful api 来调用。建议使用 Cypher 来封装调用接口。后续版本 restful api 将不再进行更新维护，只会保留登录、登出、刷新 token、cypher 调用这几个常见的 api。

English:
Q: What programming languages ​​does the client currently support, and does it support node js?
A: Currently, the main supported programming languages ​​are c++, python, java; node js is currently not supported. Users who use node as their main development language can use the restful api provided by tugraph to call. It is recommended to use Cypher to encapsulate the calling interface. Subsequent versions of the restful api will no longer be updated and maintained, and only common apis such as login, logout, refresh token, and cypher calls will be retained.

**Comment by KingSkyLi** [comment]
中文：
Q：python client 是否支持 pip install？client 在哪里进行引用？
A：目前 python client 不支持 pip 进行安装。client 在目录[https://github.com/TuGraph-db/tugraph-db/tree/master/src/client](https://github.com/TuGraph-db/tugraph-db/tree/master/src/client)。

English：
Q: Does python client support pip install? Where is the client referenced?
A: Currently the python client does not support pip for installation. client is in the directory [https://github.com/TuGraph-db/tugraph-db/tree/master/src/client](https://github.com/TuGraph-db/tugraph-db/tree/master/src/client).

**Comment by zhangwh807** [comment]
中文：
Q：TuGraph c++ SDK如何获取或构建？
A：我们有一个c++版本的旧版client，后续不再主动更新。当前我们已经升级我们的client-server协议支持bolt，但缺少c++的bolt client，如感兴趣我们欢迎和我们一起共建。
旧版本C++客户端实例在：https://github.com/TuGraph-family/tugraph-db/tree/master/test/test_rpc_client/cpp/CppClientTest

English：
Q: How to get or build the TuGraph c++ SDK?
A: We have an older version of client in c++ that we no longer actively update. At present, we have upgraded our client-server protocol to support bolt, but we lack c++ bolt client. If you are interested, we welcome to build with us.
The old version of c++ client instance in "https://github.com/TuGraph-family/tugraph-db/tree/master/test/test_rpc_client/cpp/CppClientTest"

**Comment by zhangwh807** [comment]
中文：
Q：Rust如何来访问Tugraph Server呢？
A：现在TuGraph的客户端统一都走bolt，目前是支持rust的，https://github.com/TuGraph-family/tugraph-db/blob/master/demo/Bolt/rust_example.rs

English:
Q: How does Rust access the TuGraph Server?
A: Currently, all TuGraph clients use the Bolt protocol, and Rust is supported as well. For a Rust example, please visit: https://github.com/TuGraph-family/tugraph-db/blob/master/demo/Bolt/rust_example.rs

**Comment by zhangwh807** [comment]
中文：
Q：github下载的源码压缩包里没有包含deps项目的源码，导致编译时执行deps/build_deps.sh，报找不到package.json的错误。
A：目前源码包里不包含deps源码，后续会更新。先使用git拉取的方式拉取全部源码。并推荐使用tugraph编译镜像进行源码编译，从而减少编译过程中环境配置等问题。

English:
Q: The source code package downloaded from GitHub does not include the source code for the deps project, causing an error of "package.json not found" when executing deps/build_deps.sh during compilation.
A: Currently, the source package does not contain the deps source code, and it will be updated later. For now, please use Git to clone the complete source code. It is also recommended to use the tugraph compilation image for compiling the source code, which can reduce issues related to environment configuration during the compilation process.

**Comment by Yogurteer** [comment]
如何使用python调用云部署的社区版Tugraph，我在地址https://github.com/TuGraph-db/tugraph-db/tree/master/src/client中下载了TuGraphClient.py，在本地windows环境测试使用时遇到了bug如下图
<img width="1206" alt="figure1" src="https://github.com/user-attachments/assets/6f5f8fd0-d9b6-4218-a80b-77ae6c9fbe1f">

**Comment by zhangwh807** [comment]
你好！在邮件中我看不到详细的图片，方便的话可以加我钉钉：fz0-nttld8yph
------------------------------------------------------------------
发件人：Zhenpeng Zhou ***@***.***>
发送时间：2024年10月16日(星期三) 14:22
***@***.***>
***@***.***>; ***@***.***>
主　题：Re: [TuGraph-family/tugraph-db] Client QA Summary (Discussion #109)
如何使用python调用云部署的社区版Tugraph，我在地址https://github.com/TuGraph-db/tugraph-db/tree/master/src/client中下载了TuGraphClient.py，在本地windows环境测试使用时遇到了bug如下图 <https://github.com/TuGraph-db/tugraph-db/tree/master/src/client%E4%B8%AD%E4%B8%8B%E8%BD%BD%E4%BA%86TuGraphClient.py%EF%BC%8C%E5%9C%A8%E6%9C%AC%E5%9C%B0windows%E7%8E%AF%E5%A2%83%E6%B5%8B%E8%AF%95%E4%BD%BF%E7%94%A8%E6%97%B6%E9%81%87%E5%88%B0%E4%BA%86bug%E5%A6%82%E4%B8%8B%E5%9B%BE >
figure1.png (view on web) <https://github.com/user-attachments/assets/6f5f8fd0-d9b6-4218-a80b-77ae6c9fbe1f >

**Reply by Yogurteer** [reply]
已添加
